controllers/vsphereclusteridentity_controller.go

```diff
@@ -115,6 +115,12 @@ func (r clusterIdentityReconciler) Reconcile(ctx _context.Context, req reconcile
 		}
 	}()
 
+	// Add a finalizer and requeque to ensure that the secret is deleted when the identity is deleted.
+	if !ctrlutil.ContainsFinalizer(identity, infrav1.VSphereClusterIdentityFinalizer) {
+		ctrlutil.AddFinalizer(identity, infrav1.VSphereClusterIdentityFinalizer)
+		return reconcile.Result{Requeue: true}, nil
+	}
+
```

Comment:
This section should be below the deletionTimestamp check (~ l. 127).

As soon as the deletionTimestamp is set the apiserver will block adding a new finalizers (cf with other controllers)

Reply:
I think this one is still open

Reply:
Nope it isn't :/ My bad. git pull error :D